bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/gmos/FullExposureTime.scala

```diff
@@ -0,0 +1,76 @@
+package edu.gemini.spModel.gemini.gmos
+
+import edu.gemini.spModel.config2.{ Config, ItemKey }
+import edu.gemini.spModel.gemini.calunit.calibration.CalDictionary.OBS_TYPE_ITEM
+import edu.gemini.spModel.gemini.gmos.InstGmosCommon.{ NS_STEP_COUNT_PROP_NAME, NUM_NS_CYCLES_PROP, USE_NS_PROP }
+import edu.gemini.spModel.obscomp.InstConstants.{ EXPOSURE_TIME_PROP, OBSERVE_TYPE_PROP }
+
+import edu.gemini.spModel.seqcomp.SeqConfigNames.{ INSTRUMENT_CONFIG_NAME, OBSERVE_CONFIG_NAME }
+
+import java.time.Duration
+
+import scalaz._
+import Scalaz._
+
+/** Adds support for calculating and inserting "full" exposure time for GMOS,
+  * even for nod and shuffle steps.  Nod and shuffle takes a number of exposures
+  * per step (cycle count * offset position step count) so the full exposure
+  * time is the nominal instrument exposure time times the nod and shuffle count.
+  */
+object FullExposureTime {
+
+  private object keys {
+    val StepCount:        ItemKey = new ItemKey(s"$INSTRUMENT_CONFIG_NAME:$NS_STEP_COUNT_PROP_NAME")
+    val CycleCount:       ItemKey = new ItemKey(s"$INSTRUMENT_CONFIG_NAME:${NUM_NS_CYCLES_PROP.getName}")
+    val UseNs:            ItemKey = new ItemKey(s"$INSTRUMENT_CONFIG_NAME:${USE_NS_PROP.getName}")
+    val ExposureTime:     ItemKey = new ItemKey(s"$INSTRUMENT_CONFIG_NAME:$EXPOSURE_TIME_PROP")
+    val ObsType:          ItemKey = OBS_TYPE_ITEM.key
+  }
+
+  private implicit class ConfigOps(c: Config) {
+
+    import keys._
+
+    def value(k: ItemKey): Option[Object] =
+      Option(c.getItemValue(k))
+
+    def useNs: Boolean =
+     value(UseNs).map {
+       case b: java.lang.Boolean => b.booleanValue
+       case _                    => false
```

Comment:
Is a bogus value here not an error condition (and similarly with other props below)?

Reply:
That's a good point, I'll just throw an exception if there is an unexpected type in the configuration.  Likely it won't get this far anyway but it makes more sense than just ignoring the problem.